src/convertToObject.js

```diff
@@ -3,16 +3,53 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ *  (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ *  (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const cssStyles = sourceString
+    .split(';')
+    .map(str => {
+      return str
+        .split('\n')
+        .filter(item => item !== '')
+        .map((item, i) => {
+          return item
+            .split(' ')
+            .filter(el => el !== '')
+            .map(el => {
+              return el
+                .split(':')
+                .filter(elWithoutEmptyString => elWithoutEmptyString !== '');
+            })
+            .filter(el => el.length !== 0)
+            .map(el => el[0]);
+        })
+        .filter(el => el.length !== 0);
+    })
+    .filter(el => el.length !== 0)
+    .map(arr => arr[0]);
+
+  return createObjectStyle(cssStyles);
+}
+
+function createObjectStyle(cssStyles = []) {
+  const objStyles = {};
+
+  cssStyles.forEach(style => {
+    const [key, ...values] = style;
+
+    objStyles[key] = values.join(' ');
+  });
+
```

Comment:
also, you can try to simplify your code without helper function, just using the `reduce` method